xbmc/Application.cpp

```diff
@@ -1897,7 +1897,10 @@ void CApplication::Render()
     g_Windowing.FinishPipeline();
   }
 
-  g_graphicsContext.Flip(hasRendered);
+  // TODO: find better solution
+  // if video is rendered to a separate layer, we should not block this thread
+  if (!m_pPlayer->IsRenderingVideoLayer() || hasRendered)
+    g_graphicsContext.Flip(hasRendered);
 
```

Comment:
@FernetMenta This change causes RenderUpdate() on AML to be called hundreds of times for each frame causing high CPU usage. It seems changing the above fragment to
```
if (m_pPlayer->IsRenderingVideoLayer() || hasRendered)
  g_graphicsContext.Flip(hasRendered);
```
fixes high CPU usage, but I'm not sure if this is correct solution.